src/main/scala/org/broadinstitute/dsde/firecloud/utils/EntityMatrix.scala

```diff
@@ -0,0 +1,114 @@
+package org.broadinstitute.dsde.firecloud.utils
+
+import org.broadinstitute.dsde.firecloud.core.GetEntitiesWithType.EntityWithType
+import spray.json.JsValue
+
+import scala.collection.mutable
+
+class EntityMatrix {
```

Comment:
What about this:
```scala
object EntityMatrix {
  def makeTsvString(entities: List[EntityWithType], entityType: String) = {
    ...
  }
}
```
I think it would be more Scala-y to not store mutable values like headers and rows.